File: MarvelousMashupTeam16/Assets/Scripts/DataClasses/Comms/Objects/InfinityStone.cs
public class InfinityStone : Entities, IFieldContent, IID
{
    
    public const int GREEN = 1; // time
    public const int YELLOW = 2; // mind
    public const int PURPLE = 3; // power
    public const int BLUE = 4; // space
    public const int RED = 5; // reality
    public const int ORANGE = 6; // soul


    public int stone;
    public int cooldown;
    public int defaultcooldownTime;

    /**
     * the constructor of the InfinityStone-Class. Creates a InfinityStone-MessageObject.
     *
     * @author Sarah Engele
     *
     * @param ID the ID of this specific InfinityStone (every InfinityStone has an ID,
     *           see also in IDs class)
     * @param position the position where the InfinityStone is placed on the map
     */

    public InfinityStone(int ID, int[] position) : base(EntityType.InfinityStone, ID, position){
    }

    /**
     * This methode compares two entities (Infinity Stones) with each other
     *
     * @author Sarah Engele
     *
     * @param o an object (most likely an Infinity Stone) that is going to be compared to the Infinity Stone
     * @return boolean which is true when both objects are equal (entityType, ID, position)
     */
    public override bool Equals(object o) {
        return base.Equals(o);
    }
    
    public override string ToString()
    {
        return (stone == GREEN ? "Green" :
                   stone == YELLOW ? "Yellow" :
                   stone == PURPLE ? "Purple" :
                   stone == BLUE ? "Blue" :
                   stone == RED ? "Red" :
                   stone == ORANGE ? "Orange" : "Undefined")
               + " InfinityStone";
    }
    
}

File: MarvelousMashupServerTeam23/src/main/java/communication/messages/objects/InfinityStone.java
package communication.messages.objects;

import communication.messages.enums.EntityType;

/**
 *  Message-Class, which is responsible to describe the Entity Infinity Stone which can be placed on
 *  the map. Infinity Stones are only called entities if they are placed on the map. If a character has
 *  an Infinity Stone placed in his or her inventory, the Infinity Stone is not to be found as an entity,
 *  but inside the inventory of an entity.
 *  Infinity Stones are needed e.g. for GameStateEvents
 *
 *  @author Sarah Engele
 *
 */

public class InfinityStone extends Entities{



    /**
     * the constructor of the InfinityStone-Class. Creates a InfinityStone-MessageObject.
     *
     * @author Sarah Engele
     *
     * @param ID the ID of this specific InfinityStone (every InfinityStone has an ID,
     *           see also in IDs class)
     * @param position the position where the InfinityStone is placed on the map
     */

    public InfinityStone(int ID, int[] position){

        super(EntityType.InfinityStone, ID, position);

    }

    /**
     * This methode compares two entities (Infinity Stones) with each other
     *
     * @author Sarah Engele
     *
     * @param o an object (most likely an Infinity Stone) that is going to be compared to the Infinity Stone
     * @return boolean which is true when both objects are equal (entityType, ID, position)
     */
    @Override
    public boolean equals(Object o) {
        return super.equals(o);
    }
}
